### DISCUSSION TITLE
Renaming `master` to `main`

### DISCUSSION THREAD

**Discussion by portante** [distributed-system-analysis/pbench#2080] [General]
Hello folks!

What do folks think about renaming `master` to `main`?

Recently, I mistakenly merged a few PRs where the commits were not squashed, and some of the first lines were not <= 50 characters.

I have already created a proposed `main` branch with those fixes applied, and saved the existing `master` branch as `master-2021.01.11`.

Thoughts?

**Comment by ndokos** [comment]
Has the default been changed in git? or is it just Github doing that at the moment? In any case, let's do it.

**Comment by dbutenhof** [comment]
With what ultimate purpose, really? `master` is standard GitHub convention, and I think widely expected, so this seems like extra communication required. Is there really a need for this? It sounds a bit like obfuscation for the sake of being different, but if there's some master (heh, heh) purpose I'm not getting, I'm willing to entertain the notion.

**Reply by ndokos** [reply]
Github has changed the default to main: https://www.zdnet.com/article/github-to-replace-master-with-main-starting-next-month/

**Reply by dbutenhof** [reply]
OK, fascinating; somehow I completely missed that, even though I know Red Hat was having similar heritage terminology discussions.

So, yes, let's do it!

**Comment by portante** [comment]
FYI - We have switch over to use `main` for the "main" branch of pbench.  All PRs have been updated to be based on that branch name.

In your own local repository, perform the following:

```
git remote update
git checkout -b main upstream/main
git push origin main
for br in <list of my local branches>; do
    git branch --set-upstream-to=upstream/main ${br}
done
```

**Reply by RobertKrawitz** [reply]
Not working for me:

```$ git remote update
Fetching origin
Fetching upstream
$ git checkout -b upstream/main main
fatal: 'main' is not a commit and a branch 'upstream/main' cannot be created from it```

**Comment by dbutenhof** [comment]
Pete actually got that backwards;

git checkout -b main upstream/main

**Reply by portante** [reply]
I did?  Where?  ;)

Fixed.
